pkg/apis/resource/types.go

```diff
@@ -0,0 +1,466 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resource
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// Claim is created by users to describe which resources they need.
+// Its status tracks whether the resource has been allocated and what the
+// resulting attributes are.
+//
+// This is an alpha type and requires enabling the DynamicResourceAllocation
+// feature gate.
+type Claim struct {
+	metav1.TypeMeta
+
+	// The driver must set a finalizer here before it attempts to allocate
```

Comment:
I don't think the comments on ObjectMeta will be seen nor do I think it's the right place to document how drivers are supposed to operate?

Reply:
verify-description.sh wants some kind of comment here. Putting something useful seemed better than "Standard object metadata" because at least developers will see it when reading the go doc description.

Reply:
Requiring a comment on metadata seems dumb to me, but I still don't think this is the place to be instructing driver authors.

Reply:
By “driver” my first thought would be “device plugin” - we shouldn't set a puzzle for readers to solve.

Reply:
I'll remove all of the ObjectMeta comments. As pointed out above, it's not going to show up in the API documentation and developers should better read the KEP to understand how this works.